server/reload.go

```diff
@@ -705,6 +705,46 @@ func (s *Server) Reload() error {
 	if err := s.reloadOptions(curOpts, newOpts); err != nil {
 		return err
 	}
+
+	s.mu.Lock()
+	disconnectClients := []*client{}
+	checkClients := func(kind int, clients map[uint64]*client, set PinnedCertSet) {
+		for _, c := range clients {
+			if (c.kind == kind || (c.kind == CLIENT && c.clientType() == kind)) && !c.matchesPinnedCert(set) {
```

Comment:
I had a moment when I thought that this was broken because `NATS`, `MQTT`, etc.. are separate constants that would clash would the actual c.kind (CLIENT, ROUTER, etc..). But since you call `checkClients()` with specific maps that contain only the actual kind, it's fine I guess...

And we are lucky that CLIENT is 0 and NATS starts at 1. But this code is a bit risky. I would revisit. Also make it a separate function (especially if we change it to be a bit more safe, likely with more code) instead of embedded here.

Reply:
processing them separately now.
I moved all that code into a separate function.
Keeping the utility closure.